File: app/src/main/java/fr/piotr/dismoitoutsms/util/MessageBox.java
package fr.piotr.dismoitoutsms.util;

import android.app.AlertDialog;
import android.content.Context;
import android.content.DialogInterface;
import fr.piotr.dismoitoutsms.R;

public class MessageBox {

	public static void confirm(Context context, String title, String message, final Runnable ok,
			final Runnable ko) {
		AlertDialog.Builder builder = new AlertDialog.Builder(context);
		builder.setTitle(title);
		builder.setMessage(message);
		builder.setPositiveButton(context.getString(R.string.oui),
				(dialog, which) -> {
                    if (ok != null) {
                        ok.run();
                    }
                });
		builder.setNegativeButton(context.getString(R.string.non),
				(dialog, which) -> {
                    if (ko != null) {
                        ko.run();
                    }
                });
		AlertDialog dialog = builder.create();
		dialog.show();
	}

}

File: app/src/main/java/fr/piotr/dismoitoutsms/util/MessageBox.kt
package fr.piotr.dismoitoutsms.util

import android.app.AlertDialog
import android.util.Log
import fr.piotr.dismoitoutsms.DisMoiToutSmsApplication
import fr.piotr.dismoitoutsms.R

object EmptyRunnable : Runnable {

    override fun run() {
        Log.d("EmptyRunnable", "nothing to do")
    }

}

object MessageBox {

    fun confirm(title: String = "", message: String,
                ok: Runnable = EmptyRunnable, ko: Runnable = EmptyRunnable) {
        val context = DisMoiToutSmsApplication.INSTANCE.applicationContext
        val builder = AlertDialog.Builder(context)
        builder.setTitle(title)
        builder.setMessage(message)
        builder.setPositiveButton(context.getString(R.string.oui)) { _, _ -> ok.run() }
        builder.setNegativeButton(context.getString(R.string.non)) { _, _ -> ko.run() }
        val dialog = builder.create()
        dialog.show()
    }

}
